scenarios/validation_receipts/src/main.rs

```diff
@@ -61,82 +58,81 @@ fn agent_setup(
 fn agent_behaviour(
     ctx: &mut AgentContext<HolochainRunnerContext, HolochainAgentContext<ScenarioValues>>,
 ) -> HookResult {
-    let reporter = ctx.runner_context().reporter();
-    let agent = ctx.get().cell_id().agent_pubkey().clone().to_string();
-
-    let action_hash: ActionHash = call_zome(
-        ctx,
-        "crud",
-        "create_sample_entry",
-        "this is a test entry value",
-    )?;
+    // check if pending action is empty, if so create a new entry
+    if ctx.get().scenario_values.pending_actions.is_empty() {
+        let action_hash: ActionHash = call_zome(
+            ctx,
+            "crud",
+            "create_sample_entry",
+            "this is a test entry value",
+        )?;
+
+        ctx.get_mut()
+            .scenario_values
+            .pending_actions
+            .insert(action_hash, HashMap::new());
+    }
 
-    ctx.get_mut()
+    // get the first pending action
+    let action_hash = ctx
+        .get()
         .scenario_values
         .pending_actions
-        .insert(action_hash, HashMap::new());
+        .keys()
+        .next()
```

Comment:
Heads up that keys() isn't ordered so if there are multiple this can pick a different key each time. I assume that isn't what we want.